nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,438 @@
+    """
+
+    # resample stat map
+    stat_map_img = resample_to_img(stat_map_img, bg_img,
+                                       interpolation=resampling_interpolation)
+    mask_img = resample_to_img(mask_img, bg_img, fill=1,
+                                   interpolation='nearest')
+
+    return stat_map_img, mask_img
+
+
+def _json_sprite(shape, affine, vmin, vmax, cut_slices, black_bg=False,
+                 opacity=1, draw_cross=True, annotate=True, title=None,
+                 colorbar=True):
+    """ For internal use.
+        Create a json-like structure, with all the brainsprite parameters.
+    """
+
+    # Set color parameters
+    if black_bg:
+        cfont = '#FFFFFF'
+        cbg = '#000000'
+    else:
+        cfont = '#000000'
+        cbg = '#FFFFFF'
+
+    # Deal with limitations of json dump regarding types
+    if type(vmin).__module__ == 'numpy':
+        vmin = vmin.tolist()  # json does not deal with numpy array
+    if type(vmax).__module__ == 'numpy':
+        vmax = vmax.tolist()  # json does not deal with numpy array
+
+    sprite_params = {'canvas': '3Dviewer',
+                     'sprite': 'spriteImg',
+                     'nbSlice': {'X': shape[0],
+                                 'Y': shape[1],
+                                 'Z': shape[2]},
+                     'overlay': {'sprite': 'overlayImg',
+                                 'nbSlice': {'X': shape[0],
+                                             'Y': shape[1],
+                                             'Z': shape[2]},
+                                 'opacity': opacity},
+                     'colorBackground': cbg,
+                     'colorFont': cfont,
+                     'crosshair': draw_cross,
+                     'affine': affine.tolist(),
+                     'flagCoordinates': annotate,
+                     'title': title,
+                     'flagValue': annotate,
+                     'numSlice': {'X': cut_slices[0],
+                                  'Y': cut_slices[1],
+                                  'Z': cut_slices[2]}}
+
+    if colorbar:
+        sprite_params['colorMap'] = {'img': 'colorMap',
+                                     'min': vmin,
+                                     'max': vmax}
+    return sprite_params
+
+
+def _html_brainsprite(sprite_params, stat_map_base64, bg_base64, cm_base64):
+    """ For internal use.
+        Fill a brainsprite html template with relevant parameters and data.
+    """
+
+    # Load javascript libraries
+    js_dir = os.path.join(os.path.dirname(__file__), 'data', 'js')
+    with open(os.path.join(js_dir, 'jquery.min.js')) as f:
+        js_jquery = f.read()
+    with open(os.path.join(js_dir, 'brainsprite.min.js')) as f:
+        js_brainsprite = f.read()
+
+    # Load the html template, and plug base64 data and meta-data
+    html = get_html_template('stat_map_template.html')
+    html = html.replace('INSERT_SPRITE_PARAMS_HERE', json.dumps(sprite_params))
```

Comment:
Shouldn't we have Python named format string, some thing like the HTML template with entries such as:
<pre>
...{sprite_params:s}... ....{bg_data:s}....
</pre>
and then in the Python do: html.format(sprite_params=sprite_params, bg_data=stat_map_base64, ...)

Reply:
it's an option, but we would have to replace all "{" by "{{" in the javascript. 

a better option would be to use c-style substitutions:
```
    'bg_data = %(BG_DATA)i' % {'BG_DATA': 123}
```

or Template : https://docs.python.org/3.4/library/string.html#template-strings :
```
    from string import Template
    Template("bg_data = $BG_DATA;").substitute({"BG_DATA": 1234})
```

if we choose one of these options I can open a PR to change it in other parts of the code ie the surface and connectome viewers, and you can stick to it in the stat map viewer

Reply:
so I have implemented the solution using Template, and I really like it. 

I now aggregate all the data into a dictionary, and that whole section of the code ends up a two liner, and a simple one at that:
```
html = get_html_template('stat_map_template.html')
html = Template(html).safe_substitute(sprite)
```

Reply:
It also forced me to use identical keys in the python code and the html template, which is a very nice bonus for readability.